### PR TITLE
Exterro1.0

### PR DESCRIPTION
**Pull request by sriram7797**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Description
New pull request to merge Exterro/AccessData Pack into the demisto/content reposiotry. With the Exterro FTK Connect automation platform, users can now leverage Cortex XSOARs security orchestration and automation capabilities to trigger the immediate capture and preservation of endpoint evidence by FTK Enterprise, FTK Central, or FTK Lab.

## Minimum version of Cortex XSOAR
- [x] 6.0.0
- [ ] 6.1.0
- [ ] 6.2.0
- [ ] 6.5.0

## Does it break backward compatibility?
   - [ ] Yes
       - Further details:
   - [x] No

## Must have
- [x] Tests
- [x] Documentation

### CODE REVIEW

**Comment by kgal-pan** [comment]
@JasBeilin - Please review https://github.com/demisto/dockerfiles/pull/8473. It's the cause of the [UT failure](https://app.circleci.com/pipelines/github/demisto/content/186826/workflows/c8348658-a7ea-4d07-a511-5be8c100b92d/jobs/561086).

**Review by YuvHayun — CHANGES_REQUESTED** [review]
Hi @sriram7797, thx for your contribution!
I left small change requests here and in the docker PR as well.
Also, I tagged @idovandijk to review your playbooks.

**Comment by sriram7797** [comment]
> Hi @sriram7797, thx for your contribution! I left small change requests here and in the docker PR as well. Also, I tagged @idovandijk to review your playbooks.

@YuvHayun i have updated the branch with suggested changes.. Can you please tell me exactly what changes required in the Dockerfile PR? (https://github.com/demisto/dockerfiles/pull/8473)

File: Packs/Exterro/Integrations/Exterro/Exterro.yml [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+      description: The filepath of the search and tag .
+    - name: Export Path
+      required: false
+      description: The path to export files.
+    outputs:
+    - contextPath: ExterroFTK.Workflow.Status
+      description: The Status of the of workflow trigger.
+      type: string
+    description: Returns a boolean value.
+  dockerimage: demisto/accessdata:1.1.0.32812
```
Comment (YuvHayun) [inline Packs/Exterro/Integrations/Exterro/Exterro.yml:66]:
```suggestion
  dockerimage: devdemisto/accessdata:1.1.0.33490
```

**Review by YuvHayun — CHANGES_REQUESTED** [review]
Hey @sriram7797!
I left some minor comments to fix.
I approved the docker file pr and I'll merge them both when we're done with this pr.
Thank you!

**Comment by sriram7797** [comment]
> Hey @sriram7797! I left some minor comments to fix. I approved the docker file pr and I'll merge them both when we're done with this pr. Thank you!

@YuvHayun Thanks. I have updated branch with suggested changes.

File: Packs/Exterro/Integrations/Exterro/Exterro.yml [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+      description: The filepath of the search and tag .
+    - name: Export Path
+      required: false
+      description: The path to export files.
+    outputs:
+    - contextPath: ExterroFTK.Workflow.Status
+      description: The Status of the of workflow trigger.
+      type: string
+    description: Returns a boolean value.
+  dockerimage: devdemisto/accessdata:1.1.0.33490
```
Comment (kgal-pan) [inline Packs/Exterro/Integrations/Exterro/Exterro.yml:65]:
```suggestion
  dockerimage: demisto/accessdata:1.1.0.33589
```

Comment (YuvHayun) [inline Packs/Exterro/Integrations/Exterro/Exterro.yml:65]:
```suggestion
  dockerimage: demisto/accessdata:1.1.0.33589
```

**Review by idovandijk — CHANGES_REQUESTED** [review]
Thank you for the contribution @sriram7797!
As someone who has been using your products I'm really happy to see content coming from Exterro, and hoping to see your API grow and the integration with XSOAR evolving more and more. The possibilities would be endless.

Regarding your PR, there are some things that I need to be clarified before we can continue.
The playbook `Exterro FTK: Initiate Memory Dump Collection` has one task, but the playbook and the task have no inputs.
Also looks like the `exterro-ftk-trigger-workflow` command needs to a schedule task to poll the workflow status - for that we use Generic Polling. Are you familiar with Generic Polling in XSOAR?
Additionally, the files have `fromversion` field of 5.0 instead of 6.5. Please use the Demisto SDK to `format` the files.

**Comment by sriram7797** [comment]
> idovandijk

Automation ID is the required input from user. Should it be received from playbook input and mapped to Automation ID field? Can't we directly input it to the Automation ID field? Other params are optional & used based on the use case for which task is used. i am not familiar with Generic polling in XSOAR.

**Sample use case:**
Playbook should be triggered based on the incident type and incident field can be mapped to the task fields (like Target IPs field of the task can be mapped to IP Address field from the incident)

File: Packs/Exterro/Integrations/Exterro/Exterro.yml [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+  type: 0
+  required: false
+script:
+  commands:    
+  - name: exterro-ftk-trigger-workflow
+    arguments:
+    - name: Automation ID
+      required: true
+      type: number
+      description: The Id of the automation workflow.
```
Comment (kgal-pan) [inline Packs/Exterro/Integrations/Exterro/Exterro.yml:41]: Should this not be the workflow ID?

```suggestion
    - name: workflow_id
      required: true
      type: number
      description: The Id of the automation workflow.
```

File: Packs/Exterro/Integrations/Exterro/Exterro.yml [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+    - name: Case Name
+      required: false
+      description: The name of the case.
```
Comment (kgal-pan) [inline Packs/Exterro/Integrations/Exterro/Exterro.yml:44]:
```suggestion
    - name: case_name
      required: false
      description: The name of the case.
```

File: Packs/Exterro/Integrations/Exterro/Exterro.yml [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+    - name: Case IDs
+      required: false
+      description: Value of caseids.
```
Comment (kgal-pan) [inline Packs/Exterro/Integrations/Exterro/Exterro.yml:47]:
```suggestion
    - name: case_ids
      required: false
      description: Value of caseids.
```

File: Packs/Exterro/Integrations/Exterro/Exterro.yml [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+    - name: Evidence Path
+      required: false
+      description: The filepath of the evidence.
```
Comment (kgal-pan) [inline Packs/Exterro/Integrations/Exterro/Exterro.yml:50]:
```suggestion
    - name: evidence_path
      required: false
      description: The filepath of the evidence.
```

File: Packs/Exterro/Integrations/Exterro/Exterro.yml [hunk 7]
```diff
@@ -0,0 +1,72 @@
...
+    - name: Target IPs
+      required: false
+      description: Targetips for the collection
```
Comment (kgal-pan) [inline Packs/Exterro/Integrations/Exterro/Exterro.yml:53]:
```suggestion
    - name: target_ips
      required: false
      description: Target IPs for the collection
```

File: Packs/Exterro/Integrations/Exterro/Exterro.yml [hunk 8]
```diff
@@ -0,0 +1,72 @@
...
+    - name: SearchandTag Path
+      required: false
+      description: The filepath of the search and tag .
```
Comment (kgal-pan) [inline Packs/Exterro/Integrations/Exterro/Exterro.yml:56]:
```suggestion
    - name: search_tag_path
      required: false
      description: The filepath of the search and tag
```

File: Packs/Exterro/Integrations/Exterro/Exterro.yml [hunk 9]
```diff
@@ -0,0 +1,72 @@
...
+    - name: Export Path
+      required: false
+      description: The path to export files.
```
Comment (kgal-pan) [inline Packs/Exterro/Integrations/Exterro/Exterro.yml:59]:
```suggestion
    - name: export_path
      required: false
      description: The path to export files.
```

**Comment by sriram7797** [comment]
@kgal-pan Any specific reason that we need to convert to lower case? current field names are validated against task in xsoar (dev environment). If i make change here, then i need to make another PR for docker. is that needed?

**Comment by kgal-pan** [comment]
> @kgal-pan Any specific reason that we need to convert to lower case?

The reason is that the arguments are required to be in snake case standard. This is mentioned in [Command Arguments](https://xsoar.pan.dev/docs/integrations/yaml-file#command-arguments) and [Contribution Design](https://xsoar.pan.dev/docs/tutorials/tut-design#command-arguments):

image

One of the reasons is that the auto-completion of arguments in CLI breaks when there's a space in the argument.

File: Packs/Exterro/Integrations/Exterro/Exterro.yml (hunk 5 above)
Comment (kgal-pan) [reply Packs/Exterro/Integrations/Exterro/Exterro.yml]: @sriram7797 

> Is this expected to be a comma-separated list of case IDs? If yes, use [isArray: true](https://xsoar.pan.dev/docs/integrations/yaml-file#command-arguments) and parse the argument using [argToList](https://xsoar.pan.dev/docs/reference/api/common-server-python#argtolist)

File: Packs/Exterro/Integrations/Exterro/Exterro.yml (hunk 7 above)
Comment (kgal-pan) [reply Packs/Exterro/Integrations/Exterro/Exterro.yml]: @sriram7797 - 

> Is this expected to be a comma-separated list of case IDs? If yes, use [isArray: true](https://xsoar.pan.dev/docs/integrations/yaml-file#command-arguments) and parse the argument using [argToList](https://xsoar.pan.dev/docs/reference/api/common-server-python#argtolist)

**Comment by kgal-pan** [comment]
> @kgal-pan Updated task name with snake case.

Thanks. See comments https://github.com/demisto/content/pull/20849#discussion_r962590262 and https://github.com/demisto/content/pull/20849#discussion_r962589998. These weren't addressed yet.

**Comment by sriram7797** [comment]
@kgal-pan yml file updated with isArray: true field for case_ids and target_ips

**Comment by sriram7797** [comment]
@kgal-pan isArray:true and argToList changes are addressed, playbook is removed from branch. Please process the docker PR (https://github.com/demisto/dockerfiles/pull/8534) to pass unit test.

File: Packs/Exterro/Integrations/Exterro/Exterro.yml [hunk 10]
```diff
@@ -0,0 +1,74 @@
...
+      description: The filepath of the search and tag .
+    - name: export_path
+      required: false
+      description: The path to export files.
+    outputs:
+    - contextPath: ExterroFTK.Workflow.Status
+      description: The Status of the of workflow trigger.
+      type: string
+    description: Returns a boolean value.
+  dockerimage: demisto/accessdata:1.1.0.33672
```
Comment (kgal-pan) [inline Packs/Exterro/Integrations/Exterro/Exterro.yml:67]:
```suggestion
  dockerimage: demisto/accessdata:1.1.0.33872
```

**Comment by kgal-pan** [comment]
@YuvHayun - Feel free to start the review

**Comment by YuvHayun** [comment]
> @YuvHayun - Feel free to start the review

I've done mine before and it seems like there weren't any big changes.
I approve it from my side, just waiting for @sriram7797 to change the docker image so I can run instance tests.

**Comment by sriram7797** [comment]
@kgal-pan @YuvHayun  Updated docker image. Please review.

File: Packs/Exterro/Integrations/Exterro/Exterro.py [hunk 11]
```diff
@@ -0,0 +1,76 @@
...
+        client.cases
+    except JSONDecodeError as exc:
+        raise RuntimeError('False API key provided to FTK Connect.', exc)
+    except DemistoException as exc:
+        raise RuntimeError('Authentication with FTK Connect failed.', exc)
+
+    return "ok"
+
+
+def main():
```
Comment (YuvHayun) [inline Packs/Exterro/Integrations/Exterro/Exterro.py:36]:
```suggestion
def main():  # pragma: no cover
```
I see the build is failing due to low coverage percentage (you need at least 70 %).
Please add this comment to the main function and add tests for the **_test_module** function.

That's all from my side.

File: Packs/Exterro/Integrations/Exterro/Exterro.py [hunk 12]
```diff
@@ -0,0 +1,76 @@
...
+    result = client.connect.trigger(**kwargs)
+    if result.get("Status") is not True:
+        raise ValueError("Failed to trigger automation workflow.", result.get("Status"))
+
+    return CommandResults(outputs_prefix='Accessdata.Workflow',
+                          outputs_key_field='Status',
+                          outputs=result)
+
+
+def _test_module(client):
```
Comment (YuvHayun) [inline Packs/Exterro/Integrations/Exterro/Exterro.py:24]:
```suggestion
def _test_module(client):  # pragma: no cover
```

**Review by kgal-pan — APPROVED** [review]

**Pull request merged**